### ISSUE TITLE
[Request] Simple guide to migrate to BS5

### ISSUE THREAD

**Issue by mateuslecchi** [nascent-africa/jetstrap#56]
A simples guide in Readme to migrate to BS5

I have been tried to updagra to BS5.

I remove all dependencies of Tailwind, BS4, Popper.js.

Install Popper js2, BS5 through NPM.

Change bootstrap. js to looks like this:

> `window._ = require('lodash');
> 
> import 'bootstrap'
> 
> window.axios = require('axios');
> 
> window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
> 
> window.axios = require('axios');
> 
> window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';`

But when a run npm install && npm run dev I got this error about the $sizes variable. 

> ` Compiled with some errors in 20.33s
> 
> ERROR in ./resources/sass/app.scss
> Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
> ModuleBuildError: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
> SassError: Undefined variable.
>    ╷
> 14 │       @each $size, $length in $sizes {
>    │                               ^^^^^^
>    ╵
>   resources/sass/app.scss 14:31                              @content
>   node_modules/bootstrap/scss/mixins/_breakpoints.scss 68:5  media-breakpoint-up()
>   resources/sass/app.scss 11:3                               root stylesheet
>     at processResult (/var/www/html/node_modules/webpack/lib/NormalModule.js:713:19)
>     at /var/www/html/node_modules/webpack/lib/NormalModule.js:819:5
>     at /var/www/html/node_modules/loader-runner/lib/LoaderRunner.js:399:11
>     at /var/www/html/node_modules/loader-runner/lib/LoaderRunner.js:251:18
>     at context.callback (/var/www/html/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
>     at /var/www/html/node_modules/sass-loader/dist/index.js:54:7
>     at Function.call$2 (/var/www/html/node_modules/sass/sass.dart.js:93948:16)
>     at _render_closure1.call$2 (/var/www/html/node_modules/sass/sass.dart.js:82175:12)
>     at _RootZone.runBinary$3$3 (/var/www/html/node_modules/sass/sass.dart.js:27558:18)
>     at _FutureListener.handleError$1 (/var/www/html/node_modules/sass/sass.dart.js:26107:19)
> 
> 1 ERROR in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)
> webpack compiled with 2 errors`
> 

I tried like 10 guide but no one works.

Can anyone help?

**Comment by tsommie** [comment]
Require bootstrap 5.*


```
// app.js

import '@popperjs/core'
import Alpine from 'alpinejs';

const bootstrap = require('bootstrap')

window.bootstrap = bootstrap

window.Alpine = Alpine;

Alpine.start();
```

### package.json should look similar to this. Pay attention to @popperjs version.
```json
"devDependencies": {
        "@popperjs/core": "^2.9.3",
        "alpinejs": "^3.2.3",
        "bootstrap": "^5.1.0"
    }
```

## Changes

Change all `form-row` to `row g-1`, `form-group` to `mb-3`, `mr-*` to `ms-*`, `ml-*` to `me-*` where they occur.

**Comment by mateuslecchi** [comment]
hey @tsommie i made the changes that you posted on app.js

I'm still getting this error:
![image](https://user-images.githubusercontent.com/45340326/129909291-73495556-4536-41a9-a50c-b734f550a924.png)

And my app.scss its this:
```
// Fonts
@import url('https://fonts.googleapis.com/css?family=Nunito');

// Variables
@import 'variables';

// Bootstrap
@import '~bootstrap/scss/bootstrap';

@each $breakpoint in map-keys($grid-breakpoints) {
  @include media-breakpoint-up($breakpoint) {
    $infix: breakpoint-infix($breakpoint, $grid-breakpoints);
    @each $prop, $abbrev in (width: w, height: h) {
      @each $size, $length in $sizes {
        .#{$abbrev}#{$infix}-#{$size} { #{$prop}: $length !important; }
      }
    }
  }
}

// Custom
@import "custom";
```

My package.json is:
```
{
    "private": true,
    "scripts": {
        "dev": "npm run development",
        "development": "mix",
        "watch": "mix watch",
        "watch-poll": "mix watch -- --watch-options-poll=1000",
        "hot": "mix watch --hot",
        "prod": "npm run production",
        "production": "mix --production"
    },
    "devDependencies": {
        "@popperjs/core": "^2.9.3",
        "alpinejs": "^2.7.3",
        "autoprefixer": "^10.1.0",
        "axios": "^0.21",
        "bootstrap": "^5.1.0",
        "laravel-mix": "^6.0.6",
        "lodash": "^4.17.19",
        "popper.js": "^1.16.1",
        "postcss": "^8.2.1",
        "postcss-import": "^12.0.1",
        "resolve-url-loader": "^3.1.4",
        "sass": "^1.38.0",
        "sass-loader": "^11.1.1"
    },
    "dependencies": {
        "tinymce": "^5.8.1"
    }
}

```

My webpack.config.js
```
const path = require('path');

module.exports = {
    resolve: {
        alias: {
            '@': path.resolve('resources/js'),
        },
    },
};


```

My webpack.mix.js:
```
const mix = require('laravel-mix');

/*
 |--------------------------------------------------------------------------
 | Mix Asset Management
 |--------------------------------------------------------------------------
 |
 | Mix provides a clean, fluent API for defining some Webpack build steps
 | for your Laravel application. By default, we are compiling the Sass
 | file for the application as well as bundling up all the JS files.
 |
 */

mix.js('resources/js/app.js', 'public/js')
    .sass('resources/sass/app.scss', 'public/css')
    .webpackConfig(require('./webpack.config'));

if (mix.inProduction()) {
    mix.version();
}

mix.copyDirectory('node_modules/tinymce/icons', 'public/node_modules/tinymce/icons');
mix.copyDirectory('node_modules/tinymce/plugins', 'public/node_modules/tinymce/plugins');
mix.copyDirectory('node_modules/tinymce/skins', 'public/node_modules/tinymce/skins');
mix.copyDirectory('node_modules/tinymce/themes', 'public/node_modules/tinymce/themes');
mix.copy('node_modules/tinymce/jquery.tinymce.js', 'public/node_modules/tinymce/jquery.tinymce.js');
mix.copy('node_modules/tinymce/jquery.tinymce.min.js', 'public/node_modules/tinymce/jquery.tinymce.min.js');
mix.copy('node_modules/tinymce/tinymce.js', 'public/node_modules/tinymce/tinymce.js');
mix.copy('node_modules/tinymce/tinymce.min.js', 'public/node_modules/tinymce/tinymce.min.js');

```

I found nothing about that variable $sizes.

**Comment by tsommie** [comment]
Remove

```
@each $breakpoint in map-keys($grid-breakpoints) {
  @include media-breakpoint-up($breakpoint) {
    $infix: breakpoint-infix($breakpoint, $grid-breakpoints);
    @each $prop, $abbrev in (width: w, height: h) {
      @each $size, $length in $sizes {
        .#{$abbrev}#{$infix}-#{$size} { #{$prop}: $length !important; }
      }
    }
  }
}
```
Also check the custom.scss file to  make sure no variables is used, then compile again, any errors you get after this is not Bootstrap or Jetstrap related.

**Comment by tsommie** [comment]
If your using alpine v2 then ignore the way I import it in my first comment. That is specific to version 3.

**Comment by tsommie** [comment]
If your using alpine v2 then ignore the way I import it in my first comment. That is specific to version 3.

**Comment by mateuslecchi** [comment]
> Remove
> 
> ```
> @each $breakpoint in map-keys($grid-breakpoints) {
>   @include media-breakpoint-up($breakpoint) {
>     $infix: breakpoint-infix($breakpoint, $grid-breakpoints);
>     @each $prop, $abbrev in (width: w, height: h) {
>       @each $size, $length in $sizes {
>         .#{$abbrev}#{$infix}-#{$size} { #{$prop}: $length !important; }
>       }
>     }
>   }
> }
> ```
> 


After I remove this ^ the compile runs flawlessly

Thank you so much @tsommie also I kept the way that you import the alpine e looks like no problem.

**Comment by tsommie** [comment]
> > Remove
> > ```
> > @each $breakpoint in map-keys($grid-breakpoints) {
> >   @include media-breakpoint-up($breakpoint) {
> >     $infix: breakpoint-infix($breakpoint, $grid-breakpoints);
> >     @each $prop, $abbrev in (width: w, height: h) {
> >       @each $size, $length in $sizes {
> >         .#{$abbrev}#{$infix}-#{$size} { #{$prop}: $length !important; }
> >       }
> >     }
> >   }
> > }
> > ```
> 
> After I remove this ^ the compile runs flawlessly
> 
> Thank you so much @tsommie also I kept the way that you import the alpine e looks like no problem.

Ok, but if you have issues with alpine that should be your first go-to. Also if your moving to alpine v3 there are changes you need to be aware of and am working on that for Jetstrap.

I pinned this issue cause I know there are folks who might want this information till I update it to Bootstrap 5. Am actually waiting on AdminLTE and the owner is actively working on that.

> Side note: I will be dropping COREUI in Jetstrap V3.

**Comment by mateuslecchi** [comment]
Roger that.

**Comment by tsommie** [comment]
This has been resolved with the release of v2.5
